tket/src/Circuit/macro_circ_info.cpp

```diff
@@ -370,16 +370,28 @@ static std::shared_ptr<b_frontier_t> get_next_b_frontier(
   }
   // Add any new bits introduced in this slice
   for (const std::pair<UnitID, Edge>& pair : u_frontier->get<TagKey>()) {
-    if (circ.get_edgetype(pair.second) == EdgeType::Quantum) continue;
-    Vertex next_v = circ.target(pair.second);
-    if (next_slice_lookup.find(next_v) == next_slice_lookup.end()) continue;
-    if (next_b_frontier->get<TagKey>().find(Bit(pair.first)) !=
-        next_b_frontier->end()) {
-      throw CircuitInvalidity("RAW hazard created in slicing");
+    switch (circ.get_edgetype(pair.second)) {
+      case EdgeType::Quantum:
+      case EdgeType::WASM: {
+        break;
+      }
+      case EdgeType::Classical:
+      case EdgeType::Boolean: {
+        Vertex next_v = circ.target(pair.second);
+        if (next_slice_lookup.find(next_v) == next_slice_lookup.end()) continue;
+        if (next_b_frontier->get<TagKey>().find(Bit(pair.first)) !=
+            next_b_frontier->end()) {
+          throw CircuitInvalidity("RAW hazard created in slicing");
```

Comment:
This should probably be a `TKET_ASSERT`?

Reply:
done in the last commit